user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/EditBatchDialog.java

```diff
@@ -0,0 +1,258 @@
+    addRow.addClassName("add-batch-row");
+
+    Button removeLastRow = new Button();
+    removeLastRow.setText("Remove Row");
+    removeLastRow.addClickListener(this::onRemoveLastRowClicked);
+    removeLastRow.addClassName("remove-batch-row");
+
+    setHeaderTitle("Update Batch");
+    setResizable(true);
+
+    Div batchControls = new Div();
+    batchControls.addClassName("batch-controls");
+    batchControls.add(batchNameField);
+
+    Span userHelp1 = new Span("You are editing samples for batch");
+    Span experimentNameText = new Span(experimentName);
+    experimentNameText.addClassName("experiment-name");
+    Span userHelp2 = new Span("and experiment");
+    batchNameText = new Span(batchName);
+    batchNameText.addClassName("batch-name");
+    Div userHelpText = new Div(userHelp1, batchNameText, userHelp2, experimentNameText);
+    userHelpText.addClassName("user-help-text");
+
+    Div spreadsheetControls = new Div();
+    spreadsheetControls.addClassName("spreadsheet-controls");
+
+    Span rowControls = new Span();
+    rowControls.addClassName("row-controls");
+    rowControls.add(addRow, removeLastRow);
+
+    Span errorText = new Span("Unspecific Error message");
+    errorText.addClassName("error-text");
+    errorText.setVisible(false);
+
+    spreadsheetControls.add(rowControls, errorText);
+
+    add(batchControls,
+        userHelpText,
+        spreadsheetControls,
+        spreadsheet);
+
+    batchNameField.focus();
+
+    spreadsheet.addValidationChangeListener(
+        validationChangeEvent -> {
+          if (validationChangeEvent.isInvalid()) {
+            errorText.setText(validationChangeEvent.getSource().getErrorMessage());
+            errorText.setVisible(true);
+          } else {
+            errorText.setVisible(false);
+          }
+        });
+
+    spreadsheet.resetRows();
+    for (SampleInfo existingSample : existingSamples) {
+      spreadsheet.addRow(existingSample);
+    }
+  }
+
+  private TextField createBatchNameField() {
+    final TextField batchNameField;
+    batchNameField = new TextField();
+    batchNameField.setLabel("Batch Name");
+    batchNameField.setPlaceholder("Please enter a name for this batch");
+    batchNameField.setRequired(true);
+    // must contain at least one non-whitespace character and no leading/tailing whitespace.
+    batchNameField.setPattern("^\\S+(.*\\S)*$");
+    batchNameField.setErrorMessage(
+        "The batch name must not be empty. It must not start nor end with whitespace.");
+    return batchNameField;
+  }
+
+  private void onBatchNameChanged(
+      ComponentValueChangeEvent<TextField, String> batchNameChangedEvent) {
+    batchNameText.setText(batchNameChangedEvent.getValue());
+  }
+
+  private void onRemoveLastRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.removeLastRow();
+  }
+
+  private void onAddRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.addEmptyRow();
+  }
```

Comment:
I found an interesting bug concerning on how the added or deleted rows are shown if the buttons are pressed. This is also happening in the register batch spreadsheet 😢 

https://github.com/qbicsoftware/data-manager-app/assets/29627977/26c35468-46c9-4821-b9e4-b0750df76b7f

Reply:
I investigated the dialog implementation before this PR and there the issue does not arise. 🤔
 

Uploading Bildschirmaufnahme 2023-11-24 um 18.00.27.mov…